src/cargo/sources/registry/local.rs

```diff
@@ -34,7 +35,11 @@ impl<'cfg> RegistryData for LocalRegistry<'cfg> {
         &self.index_path
     }
 
-    fn config(&self) -> CargoResult<Option<RegistryConfig>> {
+    fn load(&self, root: &Path, path: &str) -> CargoResult<Vec<u8>> {
```

Comment:
Why not `path: &Path`? We convert `str` to `Path` both for Local and Remote registry anyway. Those slashes `format!("{}/{}/{}", &fs_name[0..2], &fs_name[2..4], fs_name)` make me nervous :)